### ISSUE TITLE
failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange

### ISSUE THREAD

**Issue by vmarlier2** [OpsLevel/terraform-provider-opslevel#138]
Hello,

We are using the OpsLever provider version 0.8.11 and we are facing an issue where the provider is unable to propagate changes on an email address. Seems like there is a GRPC issue behind.
Here is the stack trace:

```sh
Error: Plugin did not respond
│ 
│   with opslevel_team_contact.team_email["XXXXXXXXXXXXXXXXXXXXXX"],
│   on opslevel.tf line 22, in resource "opslevel_team_contact" "team_email":
│   22: resource "opslevel_team_contact" "team_email" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵

Stack trace from the terraform-provider-opslevel_v0.8.11 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xe5b230]

goroutine 45 [running]:
github.com/opslevel/opslevel-go/v2023.(*Client).UpdateContact(0xc000125668, {0xc000806060, 0x26}, {{0x0, 0x0}, 0x0, {0xc000520108, 0x15}})
	github.com/opslevel/opslevel-go/v2023/team.go:503 +0x110
github.com/opslevel/terraform-provider-opslevel/opslevel.resourceTeamContactUpdate(0xc0002e4f50, 0xffffffffffffffff)
	github.com/opslevel/terraform-provider-opslevel/opslevel/resource_opslevel_team_contact.go:136 +0x239
github.com/opslevel/terraform-provider-opslevel/opslevel.resourceTeamContact.wrap.func3(0x1003880, {0x11a0a20, 0xc0007e5f00})
	github.com/opslevel/terraform-provider-opslevel/opslevel/common.go:28 +0x31
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).Apply(0xc0007397c0, 0xc000840460, 0xc0009b07c0, {0x11a0a20, 0xc0007e5f00})
	github.com/hashicorp/terraform-plugin-sdk@v1.17.2/helper/schema/resource.go:326 +0x5e3
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).Apply(0xc000068e80, 0xc000125900, 0x11af382, 0xf)
	github.com/hashicorp/terraform-plugin-sdk@v1.17.2/helper/schema/provider.go:294 +0x6a
github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ApplyResourceChange(0xc00009dd38, {0xc0002e4070, 0x5063c6}, 0xc0002e4070)
	github.com/hashicorp/terraform-plugin-sdk@v1.17.2/internal/helper/plugin/grpc_provider.go:895 +0x7c5
github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x1166560, 0xc00009dd38}, {0x15a8418, 0xc00083e1b0}, 0xc0002e4000, 0x0)
	github.com/hashicorp/terraform-plugin-sdk@v1.17.2/internal/tfplugin5/tfplugin5.pb.go:3305 +0x169
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000118000, {0x15ad7d0, 0xc000602d00}, 0xc00088a000, 0xc00010f950, 0x1dc6f60, 0x0)
	google.golang.org/grpc@v1.55.0/server.go:1337 +0xde7
google.golang.org/grpc.(*Server).handleStream(0xc000118000, {0x15ad7d0, 0xc000602d00}, 0xc00088a000, 0x0)
	google.golang.org/grpc@v1.55.0/server.go:1714 +0x9e7
google.golang.org/grpc.(*Server).serveStreams.func1.1()
	google.golang.org/grpc@v1.55.0/server.go:959 +0x8d
created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 44
	google.golang.org/grpc@v1.55.0/server.go:957 +0x165

Error: The terraform-provider-opslevel_v0.8.11 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

A little helping hand would be appreciated to fix this 🙏

**Comment by bin3377** [comment]
same issue. it crash on the same stacktrace when `apply` the change of team slack channel:

```
resource "opslevel_team_contact" "slack" {
  count = var.slack == "" ? 0 : 1
  team  = opslevel_team.this.id
  type  = "slack"
  name  = "Slack"
  value = var.slack
}

```

**Comment by taimoorgit** [comment]
@vmarlier2 @bin3377 Thanks for reporting the bug and sharing your config details. It looks like this bug was patched in version `0.8.12`, so we recommend upgrading to at least that version. (The newest version of our provider is `0.8.14`.) 

[Bugfix link for posterity](https://github.com/OpsLevel/terraform-provider-opslevel/compare/v0.8.11...v0.8.12#diff-df5ef110f26a122d845a3bddf72521159f10f398e0fc798bf139eb5194ea3dc7)

[Terraform registry](https://registry.terraform.io/providers/OpsLevel/opslevel/latest)

---

## Reproduced the bug on version `0.8.11`

```
$ cat main.tf
resource "opslevel_team_contact" "slack" {
  team = "team"
  type = "slack"
  name = "Public Channel"
  value = "#team-platform"
}
$ terraform apply # creates contact successfully
```

```
$ cat main.tf
resource "opslevel_team_contact" "slack" {
  team = "team"
  type = "slack"
  name = "Public Channel"
  value = "#platform" # changed this line
}
$ terraform apply # performs update
│ Error: Plugin did not respond
│ 
│   with opslevel_team_contact.slack,
│   on main.tf line 1, in resource "opslevel_team_contact" "slack":
│    1: resource "opslevel_team_contact" "slack" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-opslevel_v0.8.11 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x1030129a0]

goroutine 68 [running]:
github.com/opslevel/opslevel-go/v2023.(*Client).UpdateContact(0x140007d7230?, {0x14000141da0, 0x26}, {{0x0, 0x0}, 0x0, {0x14000771040, 0x9}})
        github.com/opslevel/opslevel-go/v2023/team.go:503 +0xc0
github.com/opslevel/terraform-provider-opslevel/opslevel.resourceTeamContactUpdate(0x140007d4f50, 0x0?)
        github.com/opslevel/terraform-provider-opslevel/opslevel/resource_opslevel_team_contact.go:136 +0x1fc
github.com/opslevel/terraform-provider-opslevel/opslevel.resourceTeamContact.wrap.func3(0x1035bf080?, {0x10375e5e0?, 0x140007c61b0?})
        github.com/opslevel/terraform-provider-opslevel/opslevel/common.go:28 +0x40
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).Apply(0x14000625720, 0x1400077b400, 0x14000757de0, {0x10375e5e0, 0x140007c61b0})
        github.com/hashicorp/terraform-plugin-sdk@v1.17.2/helper/schema/resource.go:326 +0x4cc
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).Apply(0x14000796f00, 0x1400089b8f0, 0x1030baf8f?, 0xf?)
        github.com/hashicorp/terraform-plugin-sdk@v1.17.2/helper/schema/provider.go:294 +0x6c
github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ApplyResourceChange(0x14000111d30, {0x140007d4380?, 0x140007d4310?}, 0x140007d4380)
        github.com/hashicorp/terraform-plugin-sdk@v1.17.2/internal/helper/plugin/grpc_provider.go:895 +0x674
github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x1037248e0?, 0x14000111d30}, {0x103774a40, 0x140007d62a0}, 0x140007d4310, 0x0)
        github.com/hashicorp/terraform-plugin-sdk@v1.17.2/internal/tfplugin5/tfplugin5.pb.go:3305 +0x164
google.golang.org/grpc.(*Server).processUnaryRPC(0x140001fe000, {0x103779e30, 0x14000582b60}, 0x140007d8000, 0x140002ab800, 0x103f711a0, 0x0)
        google.golang.org/grpc@v1.55.0/server.go:1337 +0xbe0
google.golang.org/grpc.(*Server).handleStream(0x140001fe000, {0x103779e30, 0x14000582b60}, 0x140007d8000, 0x0)
        google.golang.org/grpc@v1.55.0/server.go:1714 +0x7c4
google.golang.org/grpc.(*Server).serveStreams.func1.1()
        google.golang.org/grpc@v1.55.0/server.go:959 +0x84
created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 55
        google.golang.org/grpc@v1.55.0/server.go:957 +0x154

Error: The terraform-provider-opslevel_v0.8.11 plugin crashed!
...
```

I noticed that this only happens if you change `value` without changing `name`. Bug happens with `type = "email"` too.

## Reproduce attempt on `0.8.12`

```
$ terraform apply # create with same input as before, success
```

```
$ terraform apply # update with same change as before (only change value)
...
opslevel_team_contact.slack: Modifying... [id=Z2lkOi8vb3BzbGV2ZWwvQ29udGFjdC8zMzYxOQ]
opslevel_team_contact.slack: Modifications complete after 1s [id=Z2lkOi8vb3BzbGV2ZWwvQ29udGFjdC8zMzYxOQ]

Apply complete! Resources: 0 added, 1 changed, 0 destroyed.
```
